mam/api/api.c

```diff
@@ -338,9 +338,10 @@ retcode_t mam_api_create_channel(mam_api_t *const api, size_t const height,
     return ret;
   }
 
-  ERR_BIND_RETURN(mam_pk_t_set_add(&api->trusted_channel_ids,
-                                   trits_begin(mam_channel_id(&channel))),
-                  ret);
+  ERR_BIND_RETURN(
+      mam_pk_t_set_add(&api->trusted_channel_ids,
+                       (mam_pk_t *)trits_begin(mam_channel_id(&channel))),
```

Comment:
this works, but it's also wrong,
`mam_pk_t`  has a `pk` field of `trit_t [MAM_CHANNEL_ID_SIZE]` type, should populate a local `mam_pk_t` and add that to the set

Reply:
I initially used a temporary set like you said but ended up just adding a cast to save memory.
I'll add the sets back.

Reply:
tmp struct `mam_pk_t`?

Reply:
Yes

Reply:
Oops, said set, sorry